cmake/TryAppendCFlags.cmake

```
@@ -10,7 +10,13 @@ function(secp256k1_check_c_flags_internal flags output)
 
   # This avoids running a linker.
   set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
-  check_c_compiler_flag("${flags}" ${result})
+
+  # Some compilers (GCC) produce no diagnostic for -Wno-unknown-warning
+  # unless other diagnostics are being produced. Therefore, test the
+  # -Wsome-warning case instead of the -Wno-some-warning one.
```

```suggestion
  # Some compilers (GCC) produce no diagnostic for -Wno-some-warning,
  # if "some-warning" is unknown to the compiler and no other diagnostics are being produced.
  #  Therefore, test the
  # -Wsome-warning case instead of the -Wno-some-warning one.
```

Modulo paragraph formatting.

This confused me a lot because I thought `-Wno-unknown-warning` is a real flag that says "Don't warn on unknown warnings". (Clang has a real flag `-Wno-unknown-warning-option` for this!)

---

Thanks! Reworked.